tmpy/session/tm_sage.py

```diff
@@ -70,42 +66,53 @@ def compose_output(data):
             ps_contents = ps_file.read()
             ps_file.close()
 
-            return "ps: " + ps_contents
+            flush_ps(ps_contents)
         except:
             pass
-
-    if isinstance(data, SageObject):
+    elif isinstance(data, SageObject):
         try:
-            l = latex(data)
             #Replace latex arrays with matrix
-            return "latex:" + "$" +  l + "$"
+            flush_latex(latex(data))
         except:
-          return "verbatim: %s" % str(data)
-    if isinstance(data, str):
-        return 'verbatim:'+data.strip()
-    if isinstance(data, int):
-        return 'verbatim: %d' % data
-    if isinstance(data, float):
-        return 'verbatim: %f' % data
-  
-    if py_ver == 2:
-        if isinstance(data, unicode):
-            data2 = r''
-            for c in data:
-                if c not in string.printable:
-                    data2 += '\\x%x' % ord(c)
-                else:
-                    data2 += c
-            data = data2
-
-    return 'verbatim: %s' % str(data)
-
-my_globals = {}
+            flush_verbatim('%s' % str(data))
+    elif isinstance(data, str):
+        flush_verbatim(data.strip())
+    elif isinstance(data, int):
+        flush_verbatim('%d' % data)
+    elif isinstance(data, float):
+        flush_verbatim('%f' % data)
+    elif isinstance (data, PSOutDummy):
+        flush_ps (data.content)
+    elif isinstance (data, FileOutDummy):
+        if (data.content is None):
+            flush_verbatim ("")
+        else:
+            flush_file (data.content)
+    else:
+        print("other")
```

Comment:
for debugging purpose?

Reply:
Thanks for the review. Yes, I forgot to remove that one.